spec/controllers/talents_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+    end
+
+    it "renders the correct template" do
+      expect(@response).to render_template :index
+    end
+
+    it { should respond_with 200 }
+  end
+
+  describe "GET #show" do
+    before(:each) do
+      @talent = create :talent_with_sources
+      get :show, params: { id: @talent.id }
+    end
+
+    it "assigns the requested talent to @talent" do
+      expect(assigns(:talent)).to eq(@talent)
+    end
+
+    it "renders the correct template" do
+      expect(@response).to render_template :show
+    end
+
+    it { should respond_with 200 }
+  end
+
+  describe "GET #new" do
+    before(:each) do
+      get :new
+    end
+
+    it "assigns a new talent to @talent" do
+      expect(assigns(:talent)).not_to eql nil
+    end
+
+    it "renders the correct template" do
+      expect(@response).to render_template :new
+    end
+
+    it { should respond_with 200 }
+  end
+
+  describe "GET #edit" do
+    before(:each) do
+      @talent = create :talent_with_sources
+      get :edit, params: { id: @talent.id } 
+    end
+
+    it "assigns the requested talent to @talent" do
+      expect(assigns(:talent)).to eq(@talent)
+    end
+
+    it "renders the correct template" do
+      expect(@response).to render_template :edit
+    end
+
+    it { should respond_with 200 }
+  end
+
+  describe "POST #create" do
+    context "with valid attributes" do
+
+      before(:each) do
+        @talent_attrs = FactoryGirl.attributes_for(:talent_with_sources)
+        @talent_attrs[:sources_attributes] = { "1": { page: 95, book: "The new book", _destroy: false } }
+      end
+      it "creates a new talent" do
+        expect{
+          post :create, params: { talent: @talent_attrs }
+        }.to change(Talent, :count).by(1)
+      end
+      
+      it "redirects to the new talent" do
+        post :create, params: { talent: @talent_attrs }
+        expect(response).to redirect_to talents_url
+      end
+    end
+    
+    context "with invalid attributes" do
+      it "does not save the new talent" do
+        expect{
+          post :create, params: { talent: FactoryGirl.attributes_for(:talent, name: nil) }
+        }.to_not change(Talent, :count)
+      end
+      
+      it "re-renders the new method" do
+        post :create, params: { talent: FactoryGirl.attributes_for(:talent, name: nil) }
+        expect(response).to render_template :new
+      end
+    end 
+  end
+
+  describe 'PATCH #update' do
+    before(:each) do
+      @talent = create(:talent, name: "Parry", ranked: "true")
+    end
+    
+    context "valid attributes" do
+      it "located the requested @talent" do
+        patch :update, params: { id: @talent, talent: FactoryGirl.attributes_for(:talent, name: "Second Wind", ranked: "false") }
+        expect(assigns(:talent)).to eq(@talent)      
+      end
+    
+      it "changes @talent's attributes" do
+        patch :update, params: { id: @talent, talent: FactoryGirl.attributes_for(:talent, name: "Second Wind", ranked: "false") }
```

Comment:
So ranked is a `string` attribute that saves "false"? Why not use a boolean instead? 🤔